### PR TITLE
Fix parsing of <Trans>

### PR DESCRIPTION
**Pull request by vincaslt**
I've noticed several problems with how `<Trans>` component is being parsed, and fixed them in my PR  for my own use. Maybe this or some parts of it could be merged into the parser if they are relevant for general use? I would be willing to tweak/fix some stuff in order for this to be merged, if needed.

This fixes following:
- #151 Disabling separators would still include them in the output, only development ones. No separators are included after fixes.
- #152 Changes results like `<0>{{text}}</0>` to `{{text}}` for interpolated values. It seems that `react-i18next` stopped doing that at some point, so keys generated from value are never found by `react-i18next`: https://github.com/i18next/react-i18next/blob/9a0722dfb00ee1a9980879843ad58fceef5ac4e2/src/Trans.js#L98
- Added interpolation for string literals, because that's done by `react-i18next` too. So previous result `Value{' '}Another {'Value'}` is now `Value Another Value`
- Fixed how JSX whitespace is handled according to https://reactjs.org/docs/jsx-in-depth.html#children-in-jsx  (same, done by react-i18next, because they just use react's children)

>JSX removes whitespace at the beginning and ending of a line. It also removes blank lines. New lines adjacent to tags are removed; new lines that occur in the middle of string literals are condensed into a single space.

- Some random small things I found like - fixed some leftover `it.only`, or improved usage of test command.

### CODE REVIEW

File: src/lexers/jsx-lexer.js [hunk 1]
```diff
@@ -31,6 +31,10 @@ const JSXParserExtension = {
 export default class JsxLexer extends JavascriptLexer {
   constructor(options = {}) {
     super(options)
+    
+    // TODO: this is default in react-i18next, so maybe here too?
```
Comment (vincaslt) [inline src/lexers/jsx-lexer.js]: What do you think?

**Comment by vincaslt** [comment]
Any news?

**Comment by sourceRS** [comment]
thank you for fixing the parsing of my people :p

**Comment by eddiewang** [comment]
Thanks for the fix!

**Comment by karellm** [comment]
@vincaslt Thanks for the PR and good work. I will review this shortly. 

Are you interested in becoming a maintainer for this project? I've basically stopped coding in my life and very rarely find the motivation to dive back into it. This project is still downloaded thousands of time every week so it would be a shame if it slowly died.

**Comment by vincaslt** [comment]
I could help look after the project somewhat. Although I can't dedicate a lot of time to it, as I have my daily job and other stuff to work on. Not sure what being a maintainer entails.

File: test/parser.test.js [hunk 2]
```diff
@@ -724,6 +724,56 @@ describe('parser', () => {
       i18nextParser.end(fakeFile)
     })
 
+    it('parses keeping basic elements if transSupportBasicHtmlNodes is true', (done) => {
```
Comment (karellm) [inline test/parser.test.js]: The test description is not very clear

File: src/lexers/jsx-lexer.js (hunk 1 above)
Comment (karellm) [inline src/lexers/jsx-lexer.js]: I'm fine with that, could you just remove the `// TODO` comment?

**Comment by vincaslt** [comment]
done

**Comment by karellm** [comment]
Thank you, I merged it.

**Pull request merged**